build_tools/tests/test_artifact_backend.py

```diff
@@ -0,0 +1,377 @@
+#!/usr/bin/env python
+"""Unit tests for artifact_backend.py."""
+
+import os
+import sys
+import tempfile
+import unittest
+from pathlib import Path
+from unittest import mock
+
+sys.path.insert(0, os.fspath(Path(__file__).parent.parent))
+
+from _therock_utils.artifact_backend import (
+    ArtifactBackend,
+    LocalDirectoryBackend,
+    S3Backend,
+    create_backend_from_env,
+)
+
+
+class TestLocalDirectoryBackend(unittest.TestCase):
+    """Tests for LocalDirectoryBackend."""
+
+    def setUp(self):
+        """Create a temporary directory for testing."""
+        self.temp_dir = tempfile.mkdtemp()
```

Comment:
nit: If we use pytest consistently (vs just `unittest` and running via `python test.py`), there is a `tmp_path` fixture we could use: https://docs.pytest.org/en/stable/how-to/tmp_path.html. This looks fine though, since it uses the temp dir for additional common setup.